### PR TITLE
allow advanced shadow modules

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?240834

I don't think there's a specific reason that shadow modules can't be advanced modules, it was just the most conservative design decision to make at the time. It's possible that there's some situation with advanced modules that won't work well with shadow modules...but I'm not concerned about this PR causing significant increased support, because when/if shadow modules misbehave, there's always the workaround "don't use shadow modules, make a full copy of the module you want to shadow."

@czue / @kaapstorm

### CODE REVIEW

File: corehq/apps/app_manager/views/modules.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def get_mod_dict(mod):
         }
 
     return {
-        'modules': [get_mod_dict(m) for m in app.modules if m.module_type == 'basic'],
+        'modules': [get_mod_dict(m) for m in app.modules if m.module_type != 'shadow'],
```
Comment (czue) [inline corehq/apps/app_manager/views/modules.py]: are there any concerns about other module types? (careplan, report)

**Review by czue — APPROVED** [review]
looks good. feel free to self merge if my question is not an issue.

File: corehq/apps/app_manager/views/modules.py (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/app_manager/views/modules.py]: good point, shadow modules don't work with reporting modules. not sure about care plan, but don't think it's worth the time/effort to investigate. fixed.

**Comment by proteusvacuum** [comment]
Oh wow, cool! I think this might simplify a lot of the eNikshay case list filters :)

**Comment by orangejenny** [comment]
caveat emptor!

(but as explained, i think this will work fine)

**Pull request merged**